android/MobiledgeXSDKDemo/matchingenginehelper/src/main/java/com/mobiledgex/matchingenginehelper/MatchingEngineHelper.java

```diff
@@ -583,10 +602,26 @@ private void onFindCloudlet(AppClient.FindCloudletReply closestCloudlet) {
             }).start();
         }
 
-        startEdgeEvents();
+        if (mEdgeEventsEnabled) {
+            startEdgeEvents(newConnection);
+        }
     }
 
-    private boolean getAppInstList() throws InterruptedException, ExecutionException {
+    public boolean getAppInstList() throws InterruptedException, ExecutionException {
+        Log.i(TAG, "getAppInstList mAppName="+mAppName+" mAppVersion="+mAppVersion+" mOrgName="+mOrgName);
+        try {
+            if (mRegisterClientComplete || registerClient()) {

```

Comment:
I believe correctness of this might not last past say, 24 hours expiration.

Reply:
You are right. I have replaced this logic with checking the expiration time of the cookie and doing a new registerClient if it has expired. See new `validateCookie` method.

Reply:
👍 JWT.